src/PrestaShopBundle/Form/Admin/Configure/ShopParameters/ProductPreferences/GeneralType.php

```diff
@@ -46,25 +48,76 @@ class GeneralType extends TranslatorAwareType
     public function buildForm(FormBuilderInterface $builder, array $options)
     {
         $builder
-            ->add('catalog_mode', SwitchType::class)
-            ->add('catalog_mode_with_prices', SwitchType::class)
+            ->add('catalog_mode', SwitchType::class,
+            [
+                'label' => $this->trans('Catalog mode', 'Admin.Shopparameters.Feature'),
+                'help' => $this->trans('Catalog mode disables the shopping cart on your store. Visitors will be able to browse your products catalog, but not buy them.', 'Admin.Shopparameters.Help')
+                    . '<br>' .
+                    $this->trans('Have specific needs? Edit particular groups to let them see prices or not.', 'Admin.Shopparameters.Help'),
+            ])
+            ->add('catalog_mode_with_prices', SwitchType::class, [
+                'row_attr' => [
+                    'class' => 'catalog-mode-option',
+                ],
+                'label' => $this->trans('Show prices', 'Admin.Shopparameters.Feature'),
+                'help' => $this->trans('Display product prices when in catalog mode.', 'Admin.Shopparameters.Help'),
+            ])
             ->add('new_days_number', IntegerType::class, [
                 'required' => false,
+                'label' => $this->trans('Number of days for which the product is considered \'new\'', 'Admin.Shopparameters.Feature'),
+                'constraints' => [
+                    new Type(
+                        [
+                            'value' => 'numeric',
+                            'message' => $this->trans('The field is invalid. Please enter a positive integer number.', 'Admin.Notifications.Error'),
```

Comment:
```suggestion
                            'message' => $this->trans('The field is invalid. Please enter a positive integer.', 'Admin.Notifications.Error'),
```
I have been told to favor "integer" instead of "integer number" yesterday so I'll change it when I see it. :) Sorry about that.